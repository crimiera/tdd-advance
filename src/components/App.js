import React, { Component } from 'react';
import './App.css';
import Wallet from './Wallet';
import Loot from './Loot';
class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          App
        </header>
        <main>
          <Wallet/>
          <hr/>
          <Loot/>
        </main>
        <div >Powered by <a href='https://www.coindesk.com/price'>CoinDesk</a></div>
      </div>
    );
  }
}

export default App;
