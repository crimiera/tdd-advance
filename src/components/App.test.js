import React from 'react';
import ReactDOM from 'react-dom';
import {shallow,configure} from 'enzyme';
import App from './App';

var Adapter = require('enzyme-adapter-react-16');

configure({ adapter: new Adapter() });

describe('App',()=>{
  const app = shallow(<App/>);

  it('It render the app ',()=>{
    expect(app).toMatchSnapshot();
  });

  it('contains a Wallet container',()=>{
    expect(app.find('Connect(Wallet)').exists()).toBe(true);
  })
});

