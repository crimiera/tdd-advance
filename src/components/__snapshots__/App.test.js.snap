// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App It render the app  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
        >
          App
        </header>,
        <main>
          <Connect(Wallet) />
          <hr />
          <Connect(Loot) />
        </main>,
        <div>
          Powered by 
          <a
            href="https://www.coindesk.com/price"
          >
            CoinDesk
          </a>
        </div>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "App",
          "className": "App-header",
        },
        "ref": null,
        "rendered": "App",
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(Wallet) />,
            <hr />,
            <Connect(Loot) />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "main",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Powered by ",
            <a
              href="https://www.coindesk.com/price"
            >
              CoinDesk
            </a>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Powered by ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "CoinDesk",
              "href": "https://www.coindesk.com/price",
            },
            "ref": null,
            "rendered": "CoinDesk",
            "type": "a",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
          >
            App
          </header>,
          <main>
            <Connect(Wallet) />
            <hr />
            <Connect(Loot) />
          </main>,
          <div>
            Powered by 
            <a
              href="https://www.coindesk.com/price"
            >
              CoinDesk
            </a>
          </div>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "App",
            "className": "App-header",
          },
          "ref": null,
          "rendered": "App",
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(Wallet) />,
              <hr />,
              <Connect(Loot) />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "main",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Powered by ",
              <a
                href="https://www.coindesk.com/price"
              >
                CoinDesk
              </a>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Powered by ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "CoinDesk",
                "href": "https://www.coindesk.com/price",
              },
              "ref": null,
              "rendered": "CoinDesk",
              "type": "a",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
